### PR TITLE
feat(ci): add comment spam protection and auto-lock stale issues (#751) [ON HOLD]

### PR DESCRIPTION
**Pull request by diberry**
- [x] Explore codebase and understand current state
- [x] Fix spam score auto-close threshold from `>= 7` to `>= 5` in `scripts/moderate-spam.mjs`
- [x] Update test descriptions/comments referencing `>= 7` in `test/scripts/moderate-spam.test.ts`
- [x] Add `actions/setup-node@v4` (node-version: 22) to `moderate-new-content` and `auto-lock-stale` workflow jobs
- [x] Run tests to validate changes (23/23 moderate-spam, 13/13 lock-stale)
- [x] Final code review (0 issues found)

### CODE REVIEW

**Comment by diberry** [comment]
**Files in this PR:**
- `.github/workflows/squad-comment-moderation.yml`
- `scripts/lock-stale.mjs`
- `scripts/moderate-spam.mjs`
- `test/scripts/lock-stale.test.ts`
- `test/scripts/moderate-spam.test.ts`
- `vitest.config.ts`

File: scripts/moderate-spam.mjs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      });
+      if (!commentRes.ok) throw new Error(`HTTP ${commentRes.status}`);
+      console.log(`  ✓ Posted spam comment on issue #${issue.number}`);
+    } catch (err) {
+      console.error(`  ✗ Failed to comment on issue #${issue.number}: ${err.message}`);
+    }
+
+    console.log(`Issue #${issue.number} closed as spam.`);
+  } else if (score >= 3) {
+    try {
```
[the remark this answers is not in the document]
Comment (diberry) [reply scripts/moderate-spam.mjs:172]: @copilot apply changes based on [this feedback](https://github.com/bradygaster/squad/pull/800#discussion_r3036169429)

File: .github/workflows/squad-comment-moderation.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          sparse-checkout: scripts/lock-stale.mjs
+          sparse-checkout-cone-mode: false
+
+      - name: Lock stale closed issues and PRs
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
+          LOCK_AFTER_DAYS: '30'
+          MAX_ITEMS: '50'
+        run: node scripts/lock-stale.mjs
```
[the remark this answers is not in the document]
Comment (diberry) [reply .github/workflows/squad-comment-moderation.yml:73]: @copilot apply changes based on [this feedback](https://github.com/bradygaster/squad/pull/800#discussion_r3036169438)

**Review by tamirdresher — APPROVED** [review]
Solid work. Spam scoring logic is reasonable (pattern-based + account age), scripts are testable via DI, 36 tests covering edge cases, third-party action pinned to SHA. Rate limiting delay between lock calls is a nice touch. Ship it.

**Review by tamirdresher — APPROVED** [review]
Solid work. Spam scoring logic is reasonable, scripts are testable via DI, 36 tests covering edge cases, third-party action pinned to SHA. Rate limiting delay between lock calls is a nice touch.

**Comment by diberry** [comment]
## 🔍 Squad Review — Kaylee (Engineering)

| # | Check | Status | Notes |
|---|-------|--------|-------|
| 1 | Changelog entry | ✅ | `.changeset/comment-spam-protection.md` present (no package scope — appropriate for CI/scripts-only change) |
| 2 | Squashed to 1 commit | ✅ | 1 commit |
| 3 | CI green | ✅ | All checks pass (23 moderate-spam + 13 lock-stale tests pass) |
| 4 | Copilot comments resolved | ✅ | 9 threads — all resolved (threshold fixed to >=5, setup-node added, SHA-pin and pagination concerns addressed) |
| 5 | No .squad/ files | ✅ | Clean |
| 6 | No unrelated files | ✅ | All 8 files related to spam protection — `vitest.config.ts` change (strip-shebang plugin) required for tests, `CONTRIBUTING.md` documents moderation |
| 7 | Tests for changes | ✅ | Scripts (moderate-spam.mjs, lock-stale.mjs) fully covered by test files (moderate-spam.test.ts, lock-stale.test.ts) |
| 8 | Not a duplicate/reversal | ✅ | Unique scope (closes #751) |

**Verdict:** ✅ Ready to merge

---
*Review by Squad AI team (Kaylee — Engineering) · requested by Dina Berry*
